js/components/EventHelpers.js

```diff
@@ -130,3 +130,42 @@ export function hasAddress(event) {
     return true
   } else return false
 }
+
+export function checkSameMonth(startDate, endDate) {
+  return startDate.getMonth() === endDate.getMonth()
+}
+
+export function checkSameDay(startDate, endDate) {
+  return checkSameMonth(startDate, endDate) && startDate.getDate() === endDate.getDate()
+}
+
+export function generateDate(date) {
+  if (date) {
+    return date.toLocaleDateString(undefined, { weekday: 'short', month: 'short', day: 'numeric' })
+  }
+}
+
+export function generateDates(startDate, endDate) {
+  if (endDate && !checkSameDay(startDate, endDate)) {
+    return generateDate(startDate) + " - " + generateDate(endDate) + ", " + startDate.getFullYear()
```

Comment:
I m not checking the year since I don't think there would be events during New Year day... But if they do have, pls let me know and I'll make the year check